crates/matrix-sdk/src/test_utils/events.rs

```diff
@@ -238,6 +250,72 @@ impl EventFactory {
         builder
     }
 
+    /// Create a replacement message for an existing one given its id.
+    pub fn replacement_msg(
+        &self,
+        new_content: impl Into<String>,
+        original_event_id: &EventId,
+    ) -> EventBuilder<RoomMessageEventContent> {
+        let msg_content = RoomMessageEventContent::text_plain(new_content.into());
+        let builder = self.event(msg_content.clone());
+        builder.edit(original_event_id, msg_content.into())
+    }
```

Comment:
Ok, although we'll have to pass the message content twice then and clone it, I think.

Reply:
Since we're already providing a `content` value when we get the `EventBuilder`, should we remove the `new_content` param from `EventBuilder::edit` instead and use this content?

Reply:
Ah, I see it's done that way so we can add the `* ` prefix separately...

Reply:
No, it's different: the parameter to `text_msg` for the edit event is the fallback text, while the new content for clients that support edits will be extracted from `new_content`.